fission/spec.go

```diff
@@ -388,6 +388,19 @@ func (fr *FissionResources) validate() error {
 	// we do not error on unreferenced functions (you can call a function through workflows,
 	// `fission function test`, etc.)
 
+	// Index envs, warn on functions referencing an environment for which spes does not exist
+	environments := make(map[string]bool)
+	for _, e := range fr.environments {
+		environments[fmt.Sprintf("%s:%s", e.Metadata.Name, e.Metadata.Namespace)] = false
```

Comment:
You can also add a warning, basically the inverse of the one below: `Environment X is unused.`

Reply:
I am not sure we need this. The functions depend on environment hence warning, but environment do not depend on anything specific.